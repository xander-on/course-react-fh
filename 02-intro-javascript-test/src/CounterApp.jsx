import { useState } from "react";
import PropTypes from "prop-types"

export const CounterApp =({value}) => {
  
  const [counter, setCounter] = useState(0);

  const handleAdd = () => setCounter(counter+1);

  const handleSubtract = () => counter > 0 && setCounter(counter-1);

  const handleReset = () => setCounter(0);
    
  return(
    <>
      <h1>CounterApp</h1>
      <h2> { counter } </h2>
      <button onClick={handleSubtract}>-1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleAdd}>+1</button>
    </>
    
  );
}


CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
  value: 0
}