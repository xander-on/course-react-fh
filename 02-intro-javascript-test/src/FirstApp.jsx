
import PropTypes from "prop-types"


export const FirstApp = ({title, subTitle, name}) =>
  <>
    <h1 data-testid="test-title"> {title} </h1>
    <p>{subTitle}</p>
    <p>{subTitle}</p>
    <p>{name}</p>
  </>

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  // title: 'No hay titulo',
  subTitle: 'No hay subtitulo',
  name:'Alexander Moreno'
}



