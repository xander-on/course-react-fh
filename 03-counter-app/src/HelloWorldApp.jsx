

export const HelloWorldApp = () => <h1>Hello world App</h1>