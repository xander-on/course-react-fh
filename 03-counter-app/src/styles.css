button{
  padding: 10px 20px;
  font-size: 20px;
}