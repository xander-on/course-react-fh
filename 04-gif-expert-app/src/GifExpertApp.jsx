import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['javascript']);

  const onAddCategory = (newCategory) => {
    if(categories.includes(newCategory)) return;
    setCategories([newCategory])
  }

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory 
        onNewCategory = { (e) => onAddCategory(e)}
      />

      {
        categories.map( (category) => 
          <GifGrid 
            key={category} 
            category={category}
          />
        )
      }
    </>
  )
}
