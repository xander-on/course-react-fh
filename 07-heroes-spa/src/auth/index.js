
export * from './Pages';

export * from './context';