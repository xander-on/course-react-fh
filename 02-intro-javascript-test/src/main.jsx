import React from 'react'
import ReactDOM from 'react-dom/client';
// import {HelloWorldApp} from './HelloWorldApp';
// import {FirstApp} from './FirstApp.jsx'
import { CounterApp } from './CounterApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <FirstApp title='Hola soy koku'></FirstApp> */}
    <CounterApp value={ 20 }/>
  </React.StrictMode>
);