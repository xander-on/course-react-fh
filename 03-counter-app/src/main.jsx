import React from 'react'
import ReactDOM from 'react-dom/client';
import {HelloWorldApp} from './HelloWorldApp';
import {FirstApp} from './FirstApp.jsx'
import { CounterApp } from './CounterApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <FirstApp ></FirstApp> */}
    <CounterApp></CounterApp>
  </React.StrictMode>
);