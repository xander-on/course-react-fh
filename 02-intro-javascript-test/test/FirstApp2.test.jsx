import { render, screen } from '@testing-library/react';
import { FirstApp } from "../src/FirstApp";

//una forma de hacer las pruebas de manera mas ordenada

describe('Pruebas en <FirstApp />', () => {
  const title = 'Hola soy Goku';
  const subTitle = 'Soy un subtitle';

  test('should de hacer match con el snapshot', () => {
    const { container } = render( <FirstApp title={ title }/>);
    expect( container ).toMatchSnapshot();
  });

  test('should de mostrar el mensaje "Hola, soy Goku"', () => {
    render( <FirstApp title={ title }/>);
    expect( screen.getByText(title) ).toBeTruthy();
    // screen.debug();
  });

  test('should de mostrar el titulo en un h1', () => {
    render( <FirstApp title={ title }/>);
    expect( screen.getByRole('heading', { level:1 }).innerHTML ).toContain( title );
  });

  test('should de mostar el subtitulo enviado por props', () => { 
    render(
      <FirstApp title={ title } subTitle={ subTitle }/> 
    );

    expect( screen.getAllByText(subTitle).length ).toBe(2);
  });
  
});
