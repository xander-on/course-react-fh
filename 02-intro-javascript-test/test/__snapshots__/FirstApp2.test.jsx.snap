// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> should de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy Goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Alexander Moreno
  </p>
</div>
`;
